### PR TITLE
Add types package, update tsconfig, add copyright

### PR DESCRIPTION
**Pull request by markwolff**
Add copyright comment to each source file,
Add types package to common to publish only type definitions
Updated tsconfig as per https://github.com/Azure/anps/blob/master/anps.md

### CODE REVIEW

File: JavaScript/JavaScriptSDK/SamplingScoreGenerator.ts [hunk 1]
```diff
@@ -1,4 +1,7 @@
-﻿/// <reference path="./HashCodeScoreGenerator.ts" />
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/// <reference path="./HashCodeScoreGenerator.ts" />
```
Comment (jpiyali) [inline JavaScript/JavaScriptSDK/SamplingScoreGenerator.ts]: can u check if this is being used?

File: gruntfile.js [hunk 2]
```diff
@@ -228,6 +228,21 @@ module.exports = function (grunt) {
...
+                expand: true,
+                cwd: 'AppInsightsCommon/amd/bundle',
+                src: './**/*.d.ts',
+                dest: 'AppInsightsCommon/amd/types/'
+            }
+        },
+        rename: {
+            commonBundleToTypes: {
+                files: [
+                    {src: 'AppInsightsCommon/amd/types/applicationinsights-common.d.ts', dest: 'AppInsightsCommon/amd/types/index.d.ts'}
```
Comment (jpiyali) [inline gruntfile.js]: how are we going to share this for now with Jerry?

Comment (markwolff) [reply gruntfile.js]: I included a package.json in the destination directory for publishing the types. I've already published v0.0.1

File: AppInsightsCommon/amd/types/package.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "applicationinsights-sdk-js-types",
+  "version": "0.0.1",
+  "description": "",
+  "license": "ISC"
```
Comment (jpiyali) [inline AppInsightsCommon/amd/types/package.json]: what is this?

Comment (jpiyali) [reply AppInsightsCommon/amd/types/package.json]: should the license refer to MIT license. I am not sure what is ISC

Comment (markwolff) [reply AppInsightsCommon/amd/types/package.json]: default is ISC. I switched it to MIT

**Pull request merged**